susemanager-utils/susemanager-sls/src/states/uyuni_config.py

```diff
@@ -0,0 +1,757 @@
+                        org_admin_user: str = None, org_admin_password: str = None):
+
+        remove_systems = [sys for sys in current_systems if sys not in new_systems]
+        if remove_systems:
+            __salt__['uyuni.systemgroup_add_remove_systems'](name, False, remove_systems,
+                                                             org_admin_user=org_admin_user,
+                                                             org_admin_password=org_admin_password)
+
+        add_systems = [sys for sys in new_systems if sys not in current_systems]
+        if add_systems:
+            __salt__['uyuni.systemgroup_add_remove_systems'](name, True, add_systems,
+                                                             org_admin_user=org_admin_user,
+                                                             org_admin_password=org_admin_password)
+
+    @staticmethod
+    def _get_systems_for_group(matching: str, target: str = "glob",
+                               org_admin_user: str = None, org_admin_password: str = None):
+
+        selected_minions = __salt__['uyuni.master_select_minions'](matching, target)
+        available_system_ids = __salt__['uyuni.systems_get_minion_id_map'](org_admin_user, org_admin_password)
+
+        return [
+            available_system_ids[minion_id] for minion_id in selected_minions.get('minions', [])
+            if minion_id in available_system_ids
+        ]
+
+    def manage(self, name: str, description: str, expression: str, target: str = "glob",
+               org_admin_user: str = None, org_admin_password: str = None) -> Dict[str, Any]:
+        """
+        Create or update group
+        :param name: group name
+        :param description: group description
+        :param expression: expression used to filter which minions should be part of the group
+        :param target: target type, one of the following: glob, grain, grain_pcre, pillar, pillar_pcre,
+                pillar_exact, compound, compound_pillar_exact. Default: glob.
+        :param org_admin_user: organization administrator username
+        :param org_admin_password: organization administrator password
+        :return: dict for Salt communication
+        """
+        current_group = None
+        current_systems = None
+        try:
+            current_group = __salt__['uyuni.systemgroup_get_details'](name,
+                                                                      org_admin_user=org_admin_user,
+                                                                      org_admin_password=org_admin_password)
+            current_systems = __salt__['uyuni.systemgroup_list_systems'](name,
+                                                                         org_admin_user=org_admin_user,
+                                                                         org_admin_password=org_admin_password)
+        except Exception as exc:
+            if exc.faultCode != 2201:
```

Comment:
I would suggest adding a comment to explain what this particular fault is about (or extract the constant with an intelligible name)

Reply:
I extract them to a constant. Those codes are error codes returned by XML-RPC API